code/implementation/docker/machine/src/main/java/io/cattle/platform/docker/machine/launch/WebsocketProxyLauncher.java

```diff
@@ -0,0 +1,170 @@
+package io.cattle.platform.docker.machine.launch;
+
+import static io.cattle.platform.server.context.ServerContext.*;
+
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.core.model.Credential;
+import io.cattle.platform.hazelcast.membership.ClusterService;
+import io.cattle.platform.hazelcast.membership.ClusteredMember;
+import io.cattle.platform.lock.definition.LockDefinition;
+import io.cattle.platform.service.launcher.GenericServiceLauncher;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.fluent.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class WebsocketProxyLauncher extends GenericServiceLauncher {
+
+    private static final Logger log = LoggerFactory.getLogger(WebsocketProxyLauncher.class);
+    private static final String MASTER_CONF = "master.conf";
+
+    private static final DynamicStringProperty ACCESS_LOG = ArchaiusUtil.getString("access.log");
+    private static final DynamicStringProperty API_FILTER_PROXY_CONFIG = ArchaiusUtil.getString("api.filter.proxy.config");
+    private static final DynamicStringProperty API_FILTER_PROXY_CONFIG_FILE = ArchaiusUtil.getString("api.filter.proxy.config.file");
+
+    @Inject
+    ClusterService clusterService;
+
+    String written = "start";
+
+    @Override
+    protected List<DynamicStringProperty> getReloadSettings() {
+        List<DynamicStringProperty> list = new ArrayList<DynamicStringProperty>();
+        list.add(ACCESS_LOG);
+        list.add(API_FILTER_PROXY_CONFIG);
+        return list;
+    }
+
+    protected void prepareConfigFile() throws IOException {
+        String config = API_FILTER_PROXY_CONFIG.get();
+        if (StringUtils.isBlank(config)) {
+            new File(API_FILTER_PROXY_CONFIG_FILE.get()).delete();
+        } else {
+            try(FileWriter fw = new FileWriter(API_FILTER_PROXY_CONFIG_FILE.get())) {
+                IOUtils.write(API_FILTER_PROXY_CONFIG.get(), fw);
```

Comment:
@ibuildthecloud we will have to write the JSON content of the "API_FILTER_PROXY_CONFIG" to the file here, right? I can add it later to the code once your PR is merged.